gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,436 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Provide a Feldman Cousins algorithm."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+import matplotlib.pylab as plt
+from scipy import stats
+
+__all__ = ['find_confidence_interval_gauss',
+           'find_confidence_interval_poisson'
+           'construct_confidence_belt_PDFs',
+           'get_upper_and_lower_limit',
+           'fix_upper_and_lower_limit',
+           'find_limit',
+           'find_average_upper_limit',
+           'construct_confidence_belt']
```

Comment:
I have an example file that produces plots from the Feldman Cousins paper and compares the analytical solution with my numerical code. It uses the first seven functions.

The last function could be removed, it just provides a different interface to the functionality.